src/parser/WASMParser.cpp

```diff
@@ -614,6 +617,38 @@ class WASMBinaryReader : public wabt::WASMBinaryReaderDelegate {
         m_currentFunction->pushByteCode(code);
     }
 
+    void pushByteCode(const Walrus::I32Eqz& code, WASMOpcode opcode)
+    {
+        m_lastEqzPos = m_currentFunction->currentByteCodeSize();
+        m_currentFunction->pushByteCode(code);
+    }
+
+    void pushByteCode(const Walrus::I64Eqz& code, WASMOpcode opcode)
+    {
+        m_lastEqzPos = m_currentFunction->currentByteCodeSize();
+        m_currentFunction->pushByteCode(code);
+    }
+
+    static_assert(sizeof(Walrus::I32Eqz) == sizeof(Walrus::I64Eqz), "");
+
+    bool isLastByteCodeEqz()
+    {
+        if (m_lastEqzPos + sizeof(Walrus::I32Eqz) == m_currentFunction->currentByteCodeSize()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    void deleteLastByteCode(size_t lastByteCodeSize)
```

Comment:
No need a generic delete. `deleteLastEqz` is enough which calls `resize(m_lastEqzPos)`